### ISSUE TITLE
Check if PayPal Payments is an available gateway before displaying it on Product/Cart pages (527)

### ISSUE THREAD

**Issue by mattallan** [woocommerce/woocommerce-paypal-payments#447]
### Describe the Bug

There's a bug present on stores that don't have the WC Subscriptions plugin active and have both PayPal Payments and WooCommerce Payments with the free Subscriptions feature enabled.

When the Subscriptions with WooCommerce Payments feature is enabled, WC Payments is the only gateway that works with Subscriptions purchases. This is because the free WCPay Subscriptions offering is heavily integrated with the Stripe's own billing feature.

In order to make sure no other gateways load on Subscription product pages or when the cart contains a subscription, `subscriptions-core` filters the list of available gateways to make sure only WooCommerce Payments is available, however, PayPal Payments never checks if it's available before loading on Product/Cart pages.

We [opened an issue on WC Payments](https://github.com/Automattic/woocommerce-payments/issues/3401#issuecomment-999366086) to investigate other popular gateways (including Stripe, Square and PayPal Checkout) and what we found was PayPal Payments was the only one that needed some changes to the gateway code so I'm creating this issue to address it.

**Issue**
PayPal Payments is showing up on Subscription product pages and Cart page when the cart contains a subscription.
If the customer purchases a subscription with PayPal Payments when WC Subscriptions plugin is not active, renewal orders will not work. Only WC Payments will work.

![image](https://user-images.githubusercontent.com/2275145/150709694-999086fb-7e26-43d6-ab70-d2c3c9e05fa8.png)

**Solution**

Similarly to [code inside PayPal Checkout](https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/blob/baaa74fa1d912078fe5ca73c7abf01a72deb4b2b/includes/class-wc-gateway-ppec-cart-handler.php#L307-L311), PayPal Payments should check if the gateway is available before loading its button modules on the Product and Cart Page.

i.e. Something like:
```
$available_gateways = WC()->payment_gateways->get_available_payment_gateways();

if ( ! isset( $available_gateways['ppcp-gateway'] ) ) {
    return;
}
```

### To Reproduce

1. Install and setup the latest WC Payments version (v3.6)
2. Install and setup PayPal Payments (vaulting must be enabled)
3. Make sure PayPal buttons are enabled on product and cart pages
2. Enable the free subscriptions feature in WC Payments (**Payments > Settings > Advanced > Enable Subscriptions with WooCommerce Payments**)
3. Visit a subscription product page and notice that the PayPal buttons are loading 🔴 
4. Add the subscription product to the cart
5. Visit the cart page and notice that the PayPal buttons are loading on the cart 🔴 
6. Visit the Checkout page and notice PayPal isn't loading 🟢 

I created a test site with WC Payments + PayPal Payments to showcase this issue: https://grubby-bovid.jurassic.ninja/

### Expected Behavior

When using the Subscriptions with WooCommerce Payments feature (free subscriptions feature), PayPal Payments buttons shouldn't be loading on subscription product pages, or the cart page when the cart contains a subscription product.

This can be solved by check if ppcp-gateway is in the list of available payment gateways before rendering any buttons.

### Actual Behavior

PayPal Buttons are loading on Product and Cart pages for Subscriptions when it's not an available payment method.

### Environment

 - WordPress Version
 - WooCommerce Version
 - Plugin Version
 - Browser [e.g. Chrome, Safari] and Version
 - Any other plugins installed

### Additional Details

<details><summary>System status</summary>

  ```
  <!--If applicable, paste the system status here. Please ensure you redact or remove any identifying information. -->
  ```

</details>

**Comment by mattallan** [comment]
Hey @Dinamiko 👋, I created this issue as part of a WooCommerce Payments compatibility project that our team is currently working on.

I'm unable to assign myself to this issue and I noticed you changed the title of this issue to include (527) which I assume is linked to another number in a different system you're using, however, I just wanted to let you know that I'm looking at this issue this week and will push up a PR.

Let me know if that conflicts with anything on your side :)
Thanks!
